docs/syntax_and_semantics/alignof.md

```diff
@@ -0,0 +1,40 @@
+# alignof
+
+The `alignof` expression returns an `Int32` with the ABI alignment in bytes of a given type. For example:
+
+```crystal
+alignof(Int32) # => 4
+alignof(Int64) # => 8
+
+struct Foo
+  def initialize(@x : Int8, @y : Int16)
+  end
+end
+
+@[Extern]
+@[Packed]
+struct Bar
+  def initialize(@x : Int8, @y : Int16)
+  end
+end
+
+alignof(Foo) # => 2
+alignof(Bar) # => 1
+```
+
+For [Reference](https://crystal-lang.org/api/Reference.html) types, the alignment is the same as the alignment of a pointer:
```

Comment:
**note:** Linking to the API docs is better than nothing, but we're really missing an explanation of reference types in the book.